src/services/TestResultsService.ts

```diff
@@ -38,8 +38,9 @@ export class TestResultsService {
           }
         }
         if (filters.vin) {
-          return this.testResultsDAO.getByVin(filters.vin).then((response: { Count: any; Items: any; }) => {
-            return this.applyTestResultsFilters(response, filters);
+          return this.testResultsDAO.getByVin(filters.vin).then((result) => {
+             const response: ITestResultData = {Count: result.Count, Items: result.Items};
```

Comment:
This line does nothing, as far as I can tell. Is there a reason you can't just pass result?

Reply:
Typescript was giving typecasting errors so I had to explicitly to the type casting.

Reply:
I would have gone with 
`return this.testResultsDAO.getByVin(filters.vin).then((result: ITestResultData}) => {`